fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/perception/index/objects/SpatialObject.java

```diff
@@ -13,9 +13,10 @@
  * Contact: mosaic@fokus.fraunhofer.de
  */
 
-package org.eclipse.mosaic.fed.application.ambassador.simulation.perception;
+package org.eclipse.mosaic.fed.application.ambassador.simulation.perception.index.objects;
 
 import org.eclipse.mosaic.lib.geo.CartesianPoint;
+import org.eclipse.mosaic.lib.math.Vector3d;
 
 public interface SpatialObject {
```

Comment:
Could be `abstract class SpatialObject extends Vector3d {` and contain all position related code (e.g. cartesianPoint member)

Reply:
Looking good. And I can't think of a clean way to get rid of the abstract method without breaking the builder pattern or introducing unsafe casts.